Installer ui

delvedor opened the pull request:
Hi all, I've implemented the UI proposal as seen in #26.
I basically rewrited all the browser part.
Here's some screenshots:
![screen shot 2016-09-04 at 3 53 47 pm](https://cloud.githubusercontent.com/assets/4865608/18231346/cb69cac0-72b7-11e6-9123-ca8868e06851.png)
During installation:
![screen shot 2016-09-04 at 3 46 08 pm](https://cloud.githubusercontent.com/assets/4865608/18231338/94acb902-72b7-11e6-94e0-e16c82f96fff.png)

What do you think?
Still missing few things, such as the update of the code example and the error management if the installation fails.
For every question/suggestion I'm here.

If you want to try it by yourself clone [this](https://github.com/delvedor/installer/tree/installer-ui) and run `npm start`.

___
*- Update -*
NOTE: the PR is not ready to be merged, I've opened early to get suggestions and feedback and allow you to try it. The final version will not have the testing code (see fakeInstall).

addaleax commented on browser.js:
```diff
@@ -0,0 +1,87 @@
+const getInstalledVersion = require('./lib/check-node')
+const loadVersions = require('./lib/load.js')
+// const install = require('./lib/install')
+// for testing purpose
```
Is this supposed to be this way?

delvedor replied:
@addaleax sorry, I didn't understand what do you mean, can you explain better?

addaleax replied:
@delvedor Sorry – sure! I mean, do you want the testing code here included when the PR is merged?

delvedor replied:
@addaleax nope, I leaved there only for testing, in this way if one of you wants to try my proposal can do it without installing a lot of times Node :P
In addition as writed above I have to implement the error management.
I've opened the pull request so I can get suggestions and improvements before finish all the work :)

delvedor replied:
I've updated the first post to clarify :)

addaleax replied:
@delvedor thank you! :)

mikeal commented:
LOVE the look of this :)

delvedor commented:
Hi, I've added the error message if the installations fails, here's the screenshot.
What do you think?

![screen shot 2016-09-14 at 9 25 08 am](https://cloud.githubusercontent.com/assets/4865608/18507538/15428036-7a71-11e6-9901-214f8b7ad193.png)

The next step is add more code examples.

delvedor commented:
I was wondering if could be a good idea add a `Report` button in case of the installation fails, which takes the user to this repo issues.
What do you think?

mikeal commented:
> I was wondering if could be a good idea add a Report button in case of the installation fails, which takes the user to this repo issues.

That's a great idea! But that shouldn't hold up this commit, let's get this merged as soon as you think it's ready :)

delvedor commented:
For me this first implementation of the UI is ready, I've removed the testing code and added another code example.

Later I will open an issue for the `report` button and another one for the `install` function, in my opinion the update function (the second argument) is useless, install should only do his job, everything else can be done in the callback.

If you think there are other things to change before merging, please let me know.

JoeDoyle23 commented:
LGTM. Merging in.
